

# <span style="color:darkblue">Introducción a la programación con Python </span>

## <span style="color:darkblue">Sesión:	Sesión 2 - Características generales de Python Sintaxis</span>

### <span style="color:darkblue">Duración estimada:	4 horas</span>

## <span style="color:darkblue"> Tabla de contenidos:
 
    1- Bienvenida
    
    2- Asistencia
    
    3- Dudas de la Clase anterior
    
    4- Control de Lectura
    
    5- Objetivo
    
    6- Introducción
    
    7- Funciones Integradas
    
    8- Variables 
    
    9- Control de lectura
    
    10-Función Print()
    
    12-Control de lectura
    
    13- Comentarios
    
    14-Asistencia
    
    15 - Tarea
</span>

## Objetivo:
Comprender las reglas y convenciones que se utilizan para escribir programas del lenguaje de programación de Python, aprender estas características generales de Python te ayudará a escribir código Python de manera más efectiva y a comprender mejor los programas Python escritos por otros programadores.

### Recursos

•	Sesión 2 de Curso de Python

•	Acceso a internet

•	Python Instalado en computadora

•	Visual Studio Code

### Introducción:
Es importante comprender algunas de las características generales de la sintaxis de Python para poder dominarlo en primer lugar, Python utiliza la indentación para definir bloques de código, lo que significa que la forma en que se estructura el código es crucial para su correcta ejecución. 

Además, Python tiene una gran cantidad de tipos de datos incorporados, como números, cadenas de texto y listas, lo que lo hace versátil y adecuado para una amplia variedad de aplicaciones.

Otras características importantes incluyen el uso de estructuras de control de flujo como if/else y bucles for/while, el uso de funciones para organizar y reutilizar el código, y el uso de módulos para importar funcionalidad de otros archivos de Python. 

Por último, Python tiene una sólida gestión de excepciones, lo que permite manejar errores y problemas de manera efectiva en tiempo de ejecución.

### La línea de comandos de Python

Para probar una pequeña cantidad de código en python, a veces es más rápido y fácil no escribir el código en un archivo. Esto es posible porque Python se puede ejecutar como una línea de comando.

Escriba lo siguiente en la línea de comandos:

<img src="./images/01.png" alt="">

Que escribirá "¡Hola, mundo!" en la línea de comando:

<img src="./images/02.png" alt="">

Esa es una forma fácil de trabajar código rápido de Python desde la consola. Recordemos que en nuestro curso trabajaremos con Visual Studio Code y debido a ello tendremos en cuenta el tener una carpeta donde almacenaremos nuestro código. 


In [1]:
# Ejercicio 1
resultado = max(5, 10, 3, 8)
print(resultado) 

10


In [2]:
#Ejercicio 2
resultado = round(3.14159, 2)
print(resultado)  # Resultado

3.14


In [3]:
#Ejercicio 3
resultado = min(5, 10, 3, 8)
print(resultado) 

3


### <span style="color:darkred">Python como calculadora</span>

### <span style="color:darkred">Built-in Functions - Funciones Integradas</span>

En Python tenemos muchas funciones integradas. Las funciones integradas están disponibles globalmente para que las uses sin importar módulos ni ninguna configuración adicional. Algunas de las funciones integradas más comunes son:

```print(), len(), type(), int(), float(), str(), input(), list(), dict(), min(), max(), sum(), sorted(), open(), file(), help(), and dir().```

<img src="./images/04.png" alt="">

## Print(  )

In [4]:
# Uso de la Función print()-> imprimir un mensaje en la consola 

print('Hola Mundo')

Hola Mundo


In [5]:
nombre = 'Juan'

edad = 23

print('Mi nombre es', nombre, 'y mi edad es :',edad)

Mi nombre es Juan y mi edad es : 23


## len( )

In [6]:
# len()-> nos muestra la longitud de una cadena de caracteres
mensaje = 'Buenas tardes, este es un Hola mundo'
longitud = len(mensaje)
print('La longitud del mensaje es:', longitud)

La longitud del mensaje es: 36


### Práctica: (3 minutos)

1- Definir una variable `nombre` que guarde el nombre de ustedes, `edad` que almacene la edad.

2- Imprimir `print('Mi nombre es', nombre, 'y mi edad es :',edad)`

3- encuentre la longitud de la variable `nombre`.


## type( )

In [7]:
valor = 23

mensaje = 'Buenas tardes, este es un Hola mundo'

print('El tipo de dato de la variable valor es:',type(valor))

print('El tipo de dato de la variable mensaje es' ,type(mensaje))

El tipo de dato de la variable valor es: <class 'int'>
El tipo de dato de la variable mensaje es <class 'str'>


## float( ) , int( ), str( )

In [8]:
cadena = '123'
print('El tipo de datos de la variable cadena es:',type(cadena))

cadena = int(cadena)
print('El tipo de datos de la variable cadena es:',type(cadena))

cadena + cadena

El tipo de datos de la variable cadena es: <class 'str'>
El tipo de datos de la variable cadena es: <class 'int'>


246

In [9]:
valor = 14
print('El tipo de datos de la variable valor es:',type(valor))

valor = float(valor)
print('El tipo de datos de la variable valor es:',type(valor),valor)



El tipo de datos de la variable valor es: <class 'int'>
El tipo de datos de la variable valor es: <class 'float'> 14.0


Las características principales que Python tiene son las siguientes: 

<img src="./images/03.png" alt="">

In [10]:
if True:
    print('Esta es una prueba de indentacion:', 3 + 4)

Esta es una prueba de indentacion: 7


In [11]:
if True:
print('Esta es una prueba de indentacion:', 3 + 4)

IndentationError: expected an indented block (1135244341.py, line 2)

In [ ]:
valor = 7

if valor > 5:
    print('el numero es mayor que 5')
    
    if valor % 2 == 0:
        print('El numero es par')
    else: 
        print('el numero es impar')
else:
    print('el numero no es mayor que 5')

###   <span style="color:darkred">Comentarios</span>

Se pueden añadir comentarios en el código de Python utilizando el símbolo "#" al comienzo de una línea. Los comentarios son útiles para explicar el propósito de un código y hacerlo más fácil de entender.

Los objetivos principales de los comentarios son los siguientes:

   * Los comentarios se pueden usar para explicar el código de Python.
   * Los comentarios se pueden utilizar para hacer que el código sea más legible.
   * Los comentarios se pueden usar para evitar la ejecución al probar el código

Los comentarios comienzan con un # y Python representará el resto de la línea como un comentario:

### <span style="color:darkblue">Comentarios por línea</span>

In [ ]:
# this is comment
# este es un comentario
print('Buenas tardes a todos')

In [ ]:
print('Buenas tardes a todos') # este es un comentario

In [ ]:
print('Buenas tardes a todos') 
# este es un comentario

En este caso el texto “This is a comment” se traducirá como comentario y no se tomará en cuenta para la ejecución de código de Python. 

Los comentarios se pueden colocar al final de una línea y Python ignorará el resto de la línea:


In [ ]:
print('Buenas tardes a todos') # este es un comentario

Un comentario no tiene que ser texto que explique el código, también se puede usar para evitar que Python ejecute código:

In [ ]:
# Función que recibe un nombre y al ejecutarla nos da como resultado ¡Hola, nombre!
def saludar(nombre):
    mensaje = "¡Hola, " + nombre + "!"
    print(mensaje)

saludar("Alice")

### <span style="color:darkblue">Comentarios multilínea</span>

Python realmente no tiene una sintaxis para comentarios de varias líneas.

Para agregar un comentario de varias líneas, puede insertar un #para cada línea:

In [ ]:
#Esta funcion depende de mucha informacion para explicarla
#Linea 2 de comentarios y explicacion
#linea 3 de explicacion
print('Hola Mundo')

O, no del todo como se pretendía, puede usar una cadena de varias líneas.

Dado que Python ignorará los literales de cadena que no están asignados a una variable, puede agregar una cadena de varias líneas (comillas triples) en su código y colocar su comentario dentro de ella:


In [ ]:
""" Este es un comentario de multilineas, 
aca yo escribo todo lo que quiera 
Python ignorará los literales de cadena que 
no están asignados a una variable, 
puede agregar una cadena de varias líneas (comillas triples)
"""
print('Hola Mundo')

In [ ]:
def saludos(nombre):
    """Esta funcion toma el nombre de una persona e 
    imprime un saludo dirigido a la persona"""
    print('Hola!'+nombre+'!')

saludos('Jose')

### <span style="color:darkred">Variables</span>

Python no requiere declaraciones de variables. 

Una variable se crea simplemente asignando un valor a un nombre. 

Las variables pueden contener cualquier tipo de datos, incluyendo **números, cadenas de texto y objetos complejos.**

<span style="color:darkgreen">Una variable se crea en el momento en que le asigna un valor por primera vez. </span>

In [ ]:
# Definiendo la variable x = 5
x = 5 

print(x)

In [ ]:
y = 'Jhon'

print(y)

In [ ]:
x*y

In [ ]:
3*x

In [ ]:
3*y

In [ ]:
x = 'jhon'

3*x

x = 3

3*x

### <span style="color:darkblue">Nombres de variables</span>

Una variable puede tener un nombre corto (como x e y) o un nombre más descriptivo (edad, nombre del coche, volumen_total). Reglas para las variables de Python:

* Un nombre de variable debe comenzar con una letra o el carácter de subrayado

* Un nombre de variable no puede comenzar con un número

* Un nombre de variable solo puede contener caracteres alfanuméricos y guiones bajos (Az, 0-9 y _)

* Los nombres de las variables distinguen entre mayúsculas y minúsculas (edad, Edad y EDAD son tres variables diferentes)

* Un nombre de variable no puede ser ninguna de las palabras clave de Python.

Ejemplos de nombres de variables aceptables: 

Imprimamos algunas para verificar:

In [ ]:
myvar = 3
myvar

In [ ]:
MYVAR = 4
MYVAR

In [ ]:
myVar = 'estilo camelCase'
myVar

In [ ]:
myVar2 = 'Ojalá se pueda'
myVar2

In [ ]:
my____var = 4
my____var

Ejemplos de nombres de variables no aceptados:

In [ ]:
my var = 4

In [ ]:
2myVar = 5

In [ ]:
my-Var = 7

### <span style="color:darkblue">Nombres de variables de varias palabras</span>

Los nombres de variables con más de una palabra pueden ser difíciles de leer.
Hay varias técnicas que puede utilizar para hacerlos más legibles:


### <span style="color:darkgreen">Camel Case:</span>

Se utiliza para nombrar variables en la programación donde las palabras deben estar conectadas pero aún se desea distinguir claramente las palabras individuales. 

Por ejemplo, en variables de programación en lenguajes como Java y JavaScript.

### Definición: 
    
        En Camel Case, cada palabra comienza con una letra mayúscula, excepto la primera palabra que comienza en minúscula. No se utilizan espacios ni caracteres especiales entre palabras.
        
### Ejemplo: 

```
    miVariableEjemplo

    nombreUsuario
    
    calcularImpuesto
```

### <span style="color:darkgreen">Pascal Case:</span>

 En nombres de clases y tipos de datos, como en la programación orientada a objetos. Se utiliza para nombrar clases, estructuras y tipos de datos en lenguajes como C# y Python.

### Definición: 
    
         Similar a Camel Case, pero la primera letra de cada palabra se coloca en mayúscula.
        
### Ejemplo: 

```
    MiVariableEjemplo

    NombreUsuario
    
    CrearUsuario
```

### <span style="color:darkgreen">Snake Case:</span>

Se utiliza para nombrar variables o funciones en lenguajes que son sensibles a las mayúsculas y minúsculas y que no admiten espacios en los nombres, como Python y Ruby.

### Definición: 
    
          Las palabras están separadas por guiones bajos (subrayados). Se usan solo letras minúsculas.
        
### Ejemplo: 

```
    numero_de_telefono

    cantidad_de_productos
    
    mi_variable_ejemplo
```

### <span style="color:darkgreen">Lower Case:</span>

 Se usa para nombrar variables o identificadores generales en muchos lenguajes de programación.

### Definición: 
    
          Todas las letras están en minúscula.
        
### Ejemplo: 

```
    mivariableejemplo

    cantidaddeproductos
    
    nombre
```

### <span style="color:darkgreen">Upper Case:</span>

Se usa para nombrar constantes en mayúsculas. Por ejemplo, en matemáticas, el número Pi se representa en mayúsculas para indicar que es una constante.

### Definición: 
    
          Todas las letras están en mayúscula.
        
### Ejemplo: 

```
    PI

    MAXIMO_VALOR
    
    MIVARIABLEEJEMPLO
```

### <span style="color:darkblue">Muchos valores para múltiples variables</span>

Python le permite asignar valores a múltiples variables en una línea:

In [12]:
m = 2
y = 'Naranja'
z = 14.6

print(m)
print(y)
print(z)

2
Naranja
14.6


In [13]:
x, y, z = 2 ,'Sandía', 23.7

print(x)
print(y)
print(z)

2
Sandía
23.7


In [19]:
x, y, z = 2 ,'Sandía', 23.7

print(x, y, z) # Sin Salto de línea
print(x,'\n\n', y,'\n', z) # \n salto de línea en la funcion print()


2 Sandía 23.7
2 

 Sandía 
 23.7


### <span style="color:darkblue">Un valor para múltiples variables</span>

Y puede asignar el mismo valor a múltiples variables en una línea:

In [22]:
x = y =z = 'esto es una cadena, '

print(x, y, z)

esto es una cadena,  esto es una cadena,  esto es una cadena, 


### <span style="color:darkblue">Salida de una variable</span>

La función `print()` de Python se usa a menudo para generar variables.


In [23]:
x = 'Python is awesome! : Python es asombroso!' 

print(x)

Python is awesome! : Python es asombroso!


En la función `print()`, genera múltiples variables, separadas por una coma:

In [26]:
x = 'Python is awesome!'
y = ':' 
z = 'Python es asombroso!' 

print(x,y,z)

Python is awesome! : Python es asombroso!


También puede usar el **operador +** para generar múltiples variables:

In [27]:
x = 'Python is awesome!'
y = ':' 
z = 'Python es asombroso!' 

print(x + y + z)

Python is awesome!:Python es asombroso!


Para números, el carácter **+** funciona como un operador matemático:

In [35]:
a, b, c = 1,2,3

print(a,b,c)

print(a + b + c) # actua como una suma

print('LA suma es', 2+3+5,'y la multiplicación es:', 4*5)

1 2 3
6
LA suma es 10 y la multiplicación es: 20


En la función `print()`, cuando intenta combinar una cadena y un número con el operador +, Python le dará un error:

La mejor manera de generar múltiples variables en la función `print()` es separarlas con comas, que incluso admiten diferentes tipos de datos:

In [36]:
x, y, z = 2 ,'Sandía', 23.7

print(x + y + z)

TypeError: unsupported operand type(s) for +: 'int' and 'str'

In [37]:
x, y, z = 2 ,4.5, 23.7

print(x + y + z)

30.2


In [38]:
# Enviar nombreApellido.py

#Impresión de string básica
print('Hola Mundo')

Hola Mundo


In [40]:
# Imprimir varias variables en una línea

nombre = 'José Fuentes'
edad   = 37

print('Mi nombre es:', nombre, "y tengo", edad, "años.")

Mi nombre es: José Fuentes y tengo 37 años.


In [44]:
valorHora = 17.5
totalHoras = 8

sueldoDiario = valorHora*totalHoras

print('Nombre Empleado:', nombre,'\n', 
      'El precio por hora trabajada:', valorHora,'\n', 
      'y el sueldo diario es:',sueldoDiario )


Nombre Empleado: José Fuentes 
 El precio por hora trabajada: 17.5 
 y el sueldo diario es: 140.0


### Formato de cadena f-string

In [52]:
print(f'Mi nombre es {nombre} y tengo {edad} años.\n\n')

print(f'Nombre Empleado: {nombre}\n El precio por hora trabajada: {valorHora}\n y el sueldo diario es:{sueldoDiario}' )

Mi nombre es José Fuentes y tengo 37 años.


Nombre Empleado: José Fuentes
 El precio por hora trabajada: 17.5
 y el sueldo diario es:140.0


In [53]:
#imprimir por tipo de variable (%)
#print('Mi nombre es %tipodevariable1 y tengo %tipodevariable2 años.\n\n'%(variable1,variable2)

print('Mi nombre es %s y tengo %d años.\n\n'%(nombre,edad))

Mi nombre es José Fuentes y tengo 37 años.




In [54]:
print('Mi nombre es %s y tengo %d años.\n\n'%(edad,nombre))

TypeError: %d format: a number is required, not str

In [56]:
nombre = "Alice"
edad = 30
peso = 120

print('Mi nombre es %s , tengo %d años y peso %d.\n\n'%(nombre,peso,edad))

Mi nombre es Alice , tengo 120 años y peso 30.




In [62]:
nombre = "Alice"
precioHora = 17.5
horasTrabajadas = 44

print('Mi nombre es %s ,cobro: %d por hora y trabaje %.2f\n\n'%(nombre,horasTrabajadas,precioHora))

Mi nombre es Alice ,cobro: 44 por hora y trabaje 17.50




In [80]:
print ('Mi nombre es %s y tengo %d años y peso %d \n\n' %(nombre,edad,peso))

Mi nombre es Alice y tengo 30 años y peso 120 




In [64]:
print('Mi nombre es %s ,cobro: %d por hora y trabaje %2f\n\n'%(nombre,horasTrabajadas,precioHora))

Mi nombre es Alice ,cobro: 44 por hora y trabaje 17.500000




In [63]:
print('Mi nombre es %s ,cobro: %d por hora y trabaje %.3f\n\n'%(nombre,horasTrabajadas,precioHora))

Mi nombre es Alice ,cobro: 44 por hora y trabaje 17.500




In [65]:
# tabulación en la impresión y salto de linea

print('Nombre: \t José \n Edad: \t 37')

Nombre: 	 José 
 Edad: 	 37


In [66]:
print('Nombre: \n\t José \n  Edad: \n\t 37')

Nombre: 
	 José 
  Edad: 
	 37


In [68]:
print('Hola')
print('Mundo')

Hola
Mundo
HolaMundo


In [69]:
print('Hola', end = '')
print('Mundo')

HolaMundo


In [79]:
colores = ['verde', 'amarillo', 'rojo']

print(colores[0])
print('\n')
print(colores[1])
print('\n')
print(colores[2])
print('\n')
print(colores[-2])
print('\n')
print(colores)
print('\n')
print(*colores)
print('\n')
print(*colores, sep = ',')

verde


amarillo


rojo


amarillo


['verde', 'amarillo', 'rojo']


verde amarillo rojo


verde,amarillo,rojo


In [82]:
mensaje = 'TeneMos qUe HAcer tareA'

mensajeMinuscula = mensaje.lower()
mensajeMayuscula = mensaje.upper()

print(mensajeMinuscula )

print(mensajeMayuscula )

tenemos que hacer tarea
TENEMOS QUE HACER TAREA


In [88]:
nombreEmpleado1 = 'Jose antonio fuentes Velasquez'

nombreEmpleado2 = 'jose antonio fuentes velasquez'

nombreEmpleado1 == nombreEmpleado2

nombreEmpleado1.lower() == nombreEmpleado2.lower()

print(nombreEmpleado1.lower())
print(nombreEmpleado2.lower())

jose antonio fuentes velasquez
jose antonio fuentes velasquez


### <span style="color:darkred">Tipos de datos</span>

Python es un lenguaje de programación orientado a objetos, lo que significa que todo en Python es un objeto se convierte en tipos de datos. 

Existen varios tipos de datos incorporados que se utilizan para almacenar diferentes tipos de información. Algunos de los tipos de datos más comunes en Python son los siguientes:

#### Números

En Python, hay tres tipos de números: enteros (int), números de punto flotante (float) y números complejos (complex). Los enteros son números enteros sin decimales, como 42. 

Los números de punto flotante son números con decimales, como 3.14. 

Los números complejos son números con una parte real y una parte imaginaria, como 2 + 3j.

#### Cadenas de texto

Las cadenas de texto (str) se utilizan para almacenar texto y se definen entre comillas simples o dobles. Por ejemplo, “Hola, ¡mundo!" es una cadena de texto en Python.

#### Listas

Las listas (list) se utilizan para almacenar una colección ordenada de elementos y se definen entre corchetes. Por ejemplo, [1, 2, 3, 4] es una lista en Python.

#### Tuplas

Las tuplas (tuple) son similares a las listas, pero son inmutables, lo que significa que no se pueden modificar después de su creación. 

Las tuplas se definen entre paréntesis. Por ejemplo, (1, 2, 3, 4) es una tupla en Python.

#### Diccionarios

Los diccionarios (dict) se utilizan para almacenar una colección de pares clave-valor y se definen entre llaves. Por ejemplo, {"nombre": "Juan", "edad": 30} es un diccionario en Python.

#### Conjuntos

Los conjuntos (set) se utilizan para almacenar una colección de elementos únicos y se definen entre llaves. Por ejemplo, {1, 2, 3, 4} es un conjunto en Python.


### Tipos de datos al declarar

AL declarar una variable no es necesario indicar ningún tipo en particular, e incluso pueden cambiar de tipo después de que se hayan establecido.

In [94]:
x = str(3)

y = int(3)

z = float(3)

print(4*x)

print(x,y,z)

3333
3 3 3.0


Los tipos de variable también pueden indicarse por medio de funciones que y así especificar literalmente el tipo de variable que queremos almacenar esto podrá generar una conversión de ser necesario.

Si el valor no coincide con una conversión correcta generará un error por ejemplo cambiar letras a enteras que no pueden ser cambiados. 

## Aplicando Conocimientos

### Ejercicio de aplicación:

-	Crea un archivo de Python en donde almacenaremos información personal, esta información debe ir con el tipo de dato correspondiente. 
Variables a crear:
    
o	Nombres

o	PrimerApellido

o	SegundoApellido

o	Edad

o	Nacionalidad

o	Sexo
